### ISSUE TITLE
CKV2_AWS_46 reports false positive on Lambda URL custom origins

### ISSUE THREAD

**Issue by triszt4n** [bridgecrewio/checkov#6551]
**Describe the issue**

CKV2_AWS_46 does not consider [newly introduced Lambda URL origins](https://aws.amazon.com/about-aws/whats-new/2024/04/amazon-cloudfront-oac-lambda-function-url-origins/) as a custom origin and thus only assumes if there's OAC in config, then it's S3 origin.

https://github.com/bridgecrewio/checkov/blob/main/checkov/terraform/checks/graph_checks/aws/CLoudFrontS3OriginConfigWithOAI.yaml

The check needs more 

**Examples**

```tf
resource "aws_cloudfront_distribution" "this" {
  # ...
  origin {
    origin_id                = "lambda-origin"
    domain_name              = "${aws_lambda_function_url.this.url_id}.lambda-url.${data.aws_region.current.name}.on.aws"
    origin_access_control_id = aws_cloudfront_origin_access_control.this.id

    custom_origin_config {
      http_port              = 80
      https_port             = 443
      origin_protocol_policy = "https-only"
      origin_ssl_protocols   = ["TLSv1.2"]
    }
  }
  # ...
}
```

**Version (please complete the following information):**
 - Using Checkov from GitHub Actions `bridgecrewio/checkov-action@v12`

**Additional context**

_None_

**Comment by tsmithv11** [comment]
Hey @triszt4n, thanks for reaching this out! I'm not able to reproduce the false positive from your example. This policy only flags if the `aws_cloudfront_distribution` is connected to a `aws_s3_bucket`, which your example is not. Are you suggesting it's a false negative and should flag lambdas not using OAC? This would be a new check and a good one. Please confirm and of course we'd welcome a contribution!
